### PR TITLE
Adding test install workflow for GitHub Actions pipelines

### PR DESCRIPTION
**Pull request by ShutdownRepo**
Added a workflow that makes sure volatility installs and runs as intended for all push and pull request triggers.
Will help avoid issues like https://github.com/volatilityfoundation/volatility3/issues/1002 in the future

### CODE REVIEW

**Comment by ikelos** [comment]
@npetroni   Could you please look this over and check it's all ok?

**Comment by ShutdownRepo** [comment]
Sorry for the multiple commits, but we're good to go now
The pipeline runs on a matrix strategy with multiple operating systems and multiple python versions, to make sure volatility installs and runs smoothly everywhere
Here is an example run
image

**Comment by ShutdownRepo** [comment]
Hello there, any news on this? Do you need me to change something?

**Comment by npetroni** [comment]
> Hello there, any news on this? Do you need me to change something?

It looks like this is going to fail on the macOS hosts. Would you like to try to fix it or remove those from the PR?
https://github.com/volatilityfoundation/volatility3/actions/runs/6215300225

**Comment by ShutdownRepo** [comment]
In my opinion, macOS should be removed from the pipeline as long as there is no installation guidance specific to this OS for volatility. The reason it's not building is it's missing a couple dependencies and vol3 docs don't address this. I also think vol3 will be mostly be used by win & linux users and not so much macOS. And even if macOS users want to, the best course of actions for them is probably to use a container. I'm a macOS user myself and that's what I'd probably do.
TL;DR: removing macOS from the list, could be added later on if it's in volatility3's roadmap to have a clear install process for this OS.

**Review by npetroni — APPROVED** [review]
This change works for me.
